source/math/BoolVectorList.ooc

```diff
@@ -0,0 +1,119 @@
+/*
+* Copyright (C) 2014 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+use ooc-collections
+import math
+import FloatVectorList
+
+BoolVectorList: class extends VectorList<Bool> {
+	init: func ~default {
+		this super()
+	}
+	init: func ~heap (capacity: Int){
+		super(capacity)
+	}
+	init: func ~fromVectorList (other: VectorList<Bool>) {
+		this super(other _vector)
+		this _count = other count
+	}
+	totalTrues: func -> Int {
+		result := 0
+		for (i in 0 .. this count)
+			result += (this[i] ? 1 : 0)
+		result
+	}
+	totalFalses: func -> Int {
+		this _count - totalTrues()
+	}
```

Comment:
I don't like these 2 function names, and besides, they should be properties, not functions. Here's two ideas:
```ooc
// Returns both numbers, use _ in the result variable to ignore either
	tally: func -> (Int, Int) {
		trues := 0
		for (i in 0 .. this count)
			trues += (this[i] ? 1 : 0)
		(trues, this _count - trues)
	}
```

```
// call with tally(true) to get the number of trues, tally(false) to get the number of falses
	tally: func (tallyTrues: Bool) -> Int {
		trues := 0
		for (i in 0 .. this count)
			trues += (this[i] ? 1 : 0)
		tallyTrues ? trues : this _count - trues
	}
```